From 74631780b8f1ca99d113fe7bad4100e02ab928bf Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Fri, 22 Oct 2021 12:56:56 -0700
Subject: [PATCH 19/28] iio: frequency: adrf6780: Fix
 adrf6780_spi_{read,write}()

Clang warns:

drivers/iio/frequency/adrf6780.c:117:1: error: all paths through this
function will call itself [-Werror,-Winfinite-recursion]
{
^
drivers/iio/frequency/adrf6780.c:138:1: error: all paths through this
function will call itself [-Werror,-Winfinite-recursion]
{
^
2 errors generated.

The underscore variants should be used here.

Fixes: 63aaf6d06d87 ("iio: frequency: adrf6780: add support for ADRF6780")
Link: https://github.com/ClangBuiltLinux/linux/issues/1490
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
Link: https://lore.kernel.org/r/20211022195656.1513147-1-nathan@kernel.org
---
 drivers/iio/frequency/adrf6780.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/iio/frequency/adrf6780.c b/drivers/iio/frequency/adrf6780.c
index abe8b30fceca..8255ffd174f6 100644
--- a/drivers/iio/frequency/adrf6780.c
+++ b/drivers/iio/frequency/adrf6780.c
@@ -118,7 +118,7 @@ static int adrf6780_spi_read(struct adrf6780_state *st, unsigned int reg,
 	int ret;
 
 	mutex_lock(&st->lock);
-	ret = adrf6780_spi_read(st, reg, val);
+	ret = __adrf6780_spi_read(st, reg, val);
 	mutex_unlock(&st->lock);
 
 	return ret;
@@ -139,7 +139,7 @@ static int adrf6780_spi_write(struct adrf6780_state *st, unsigned int reg,
 	int ret;
 
 	mutex_lock(&st->lock);
-	ret = adrf6780_spi_write(st, reg, val);
+	ret = __adrf6780_spi_write(st, reg, val);
 	mutex_unlock(&st->lock);
 
 	return ret;
-- 
2.33.1.637.gf443b226ca

